ApiHelper.php

```diff
@@ -138,35 +138,32 @@ public function getMonthlyPageviews( $pages, $start, $end ) {
 					$titles[] = $r['title'];
 				}
 			}
-			// Get monthly pageviews for all of the titles i.e. original page and its redirects
+			// Get monthly pageviews for all of the titles i.e. original page and its redirects using promises
+			$promises = [];
 			foreach ( $titles as $title ) {
 				$url = 'https://wikimedia.org/api/rest_v1/metrics/pageviews/per-article/en.wikipedia/all-access/user/' . rawurlencode( $title ) . '/monthly/' . $start . '/' . $end;
-				$result = json_decode( @file_get_contents( $url ), true );
-				if ( isset( $result['items'] ) ) {
-					$results[$page] += (int)$result['items'][0]['views'];
-				} else {
-					// Report missing data
-					/*
+				$promise = $client->getAsync( $url );
+				$promises[] = $promise; // Add the promise for each request to promises array
+			}
+			try {
+				$responses = GuzzleHttp\Promise\settle( $promises )->wait();
+			} catch ( Exception $e ) {
+				// Ignore since we are logging those below anyway.
```

Comment:
I wasn't sure what exception you were trying to catch. Going by the docs, that says `unwrap` will throw an exception if one of the request fails. `settle(...)->wait()` should continue on and let you handle the bad response yourself as you are doing below. Maybe the docs are wrong, but if they are right than that would make the try-catch redundant, is all.

Note also you might not want to log 404s, just because there might be a lot of them. Many small stubs, etc., have never been viewed since they were created or since the API came to fruition, so they will return a 404, which you can safely consider as 0 pageviews and not an actual error.